galvan/src/main/java/org/terracotta/testing/master/ServerProcess.java

```diff
@@ -250,83 +248,47 @@ private SimpleEventingStream buildEventingStream(VerboseOutputStream stdout) {
     eventMap.put("Moved to State[ PASSIVE-STANDBY ]", passiveReadyName);
+    eventMap.put("Started the server in diagnostic mode", diagnosticReadyName);
     eventMap.put("Restarting the server", zapEventName);
     eventMap.put("WARN", warn);
     eventMap.put("ERROR", err);
 
     // We will attach the event stream to the stdout.
     SimpleEventingStream outputStream = new SimpleEventingStream(serverBus, eventMap, stdout);
-    serverBus.on(pidEventName, new EventListener() {
-      @Override
-      public void onEvent(Event event) throws Throwable {
-        String line = event.getData(String.class);
-        Matcher m = Pattern.compile("PID is ([0-9]*)").matcher(line);
-        if (m.find()) {
-          try {
-            String pid = m.group(1);
-            ServerProcess.this.didStartWithPid(Long.parseLong(pid));
-          } catch (NumberFormatException format) {
-            Assert.unexpected(format);
-          }
-        } else {
-          // This is a little unusual, since it is a partial match, so at least log it in case something is wrong.
-          ServerProcess.this.harnessLogger.error("Unexpected PID-like line from server: " + line);
+    serverBus.on(pidEventName, event -> {
+      String line = event.getData(String.class);
+      Matcher m = Pattern.compile("PID is ([0-9]*)").matcher(line);
+      if (m.find()) {
+        try {
+          String pid = m.group(1);
+          didStartWithPid(Long.parseLong(pid));
+        } catch (NumberFormatException format) {
+          Assert.unexpected(format);
         }
+      } else {
+        // This is a little unusual, since it is a partial match, so at least log it in case something is wrong.
+        harnessLogger.error("Unexpected PID-like line from server: " + line);
       }
     });
-    serverBus.on(activeReadyName, new EventListener() {
-      @Override
-      public void onEvent(Event event) throws Throwable {
-        ServerProcess.this.didBecomeActive(true);
-      }
-    });
-    serverBus.on(passiveReadyName, new EventListener() {
-      @Override
-      public void onEvent(Event event) throws Throwable {
-        ServerProcess.this.didBecomeActive(false);
-      }
-    });
-    serverBus.on(zapEventName, new EventListener() {
-      @Override
-      public void onEvent(Event event) throws Throwable {
-        ServerProcess.this.instanceWasZapped();
+    serverBus.on(activeReadyName, event -> stateInterlock.serverBecameActive(this));
+    serverBus.on(passiveReadyName, event -> {
+      if (pid != 0) { //  zapped, don't do anything
+        stateInterlock.serverBecamePassive(ServerProcess.this);
       }
     });
-    serverBus.on(warn, (event) -> handleWarnLog(event));
-    serverBus.on(warn, (event) -> handleErrorLog(event));
+    serverBus.on(diagnosticReadyName, event -> stateInterlock.serverBecameDiagnostic(this));
+    serverBus.on(zapEventName, event -> ServerProcess.this.instanceWasZapped());
```

Comment:
Do not change this much code for lambdas.  I would just add your additional event as a lambda if desired.

Reply:
Okay, will revert.